src/test/java/com/example/cosmocats/service/ProductServiceTest.java

```diff
@@ -0,0 +1,119 @@
+package com.example.cosmocats.service;
+
+import com.example.cosmocats.domain.Product;
+import com.example.cosmocats.service.exception.ProductNotFoundException;
+import com.example.cosmocats.service.impl.ProductServiceImpl;
+import com.example.cosmocats.service.repository.ProductRepository;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest(classes = {ProductServiceImpl.class})
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Product Service Tests")
+public class ProductServiceTest {
+
+    @Autowired
+    private ProductService productService;
+    @MockitoBean
+    private ProductRepository productRepository;
+
+    @Captor
+    private ArgumentCaptor<Product> productArgumentCaptor;
+
+
+    public static Stream<Product> provideProducts() {
+        return Stream.of(
+                buildProduct("Product 1", "Some description 1", 100),
+                buildProduct("Product 2", "Some description 2", 200),
+                buildProduct("Product 3", "Some description 3", 300),
+                buildProduct("Product 4", "Some description 4", 400)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideProducts")
+    @Order(1)
```

Comment:
no need to do it with the order - I just did it for example, here it doesn't make sense as the tests are independent from each other :) but good that you added parametrized - +1 for the effort